File: frontend/src/services/wishlist_service.js
import { apiGet,apiPost, getFormattedDate } from "./util";

export async function loadAllWishlists() {
  try {
    const wishlists = await apiGet("/wishlist/all");
    return wishlists.map(w => ({
      ...w,
      created_at: getFormattedDate(w.created_at)
    }));
  } catch (error) {
    console.log(error);
    return [];
  }
}

export async function createNewWishlist(name,description){
    try{
        const response = await apiPost("/wishlist/create",{
            name:name,
            description: description,
        });
        return response;
    }catch(error){
        console.log(error);
       return null;
    }
}
export async function addUserToWishlist(user_id, wishlist_id) {
  try {
    const response = await apiPost("/wishlist/add_user", {
      user_id: user_id,
      wishlist_id: wishlist_id,
    });
    return response;
  } catch (error) {
    console.error("Error adding user to wishlist:", error);
    return null;
  }
}

File: backend/services/wishlist_service.py
from database import db
from models.user_model import User
from models.wishlist_model import Wishlist
from models.item_model import Item
    # id = db.Column(db.Integer, primary_key=True)
    # name = db.Column(db.String(120), nullable=False)
    # user_count = db.Column(db.Integer, default=0)
    # description = db.Column(db.Text, nullable=True)
    # created_at = db.Column(db.DateTime(timezone=True), server_default=func.now())

def create_wishlist(name=None, user_count=0, description=None):
    if name:
        new_wishlist = Wishlist(name=name, user_count=user_count, description=description)
        db.session.add(new_wishlist)
        db.session.commit()
        return new_wishlist
    return None

def get_all_wishlists():
    wishlists = db.session.query(Wishlist).all()
    return [w.to_dict() for w in wishlists] if wishlists else []

def add_user_to_wishlist(wishlist_id, user_id):
    wishlist = Wishlist.query.get(wishlist_id)
    user = User.query.get(user_id)

    if not wishlist or not user:
        return None
    
    if user not in wishlist.users:
        wishlist.users.append(user)
        wishlist.user_count = len(wishlist.users)
        db.session.commit()

    return wishlist

def get_wishlist_items_by_id(wishlist_id):
    items = db.session.query(Item).filter_by(wishlist_id=wishlist_id).all()
    return [item.to_dict() for item in items] if items else []
